### DISCUSSION TITLE
Unexpected results when searching by latitude and longitude

### DISCUSSION THREAD

**Discussion by usul0184** [photoprism/photoprism#5101] [Q&A]
Great project!  I've been experiencing some trouble when searching by latitude (lat) or longitude (lng).  The results returned appear to be approximated in a way that doesn't make any obvious sense to me.  For example, I have an image with latitude 33.382225.  It comes up as a result, though, for searches ranging from lat:33.29 through lat:33.39.  I'm not sure why such a large range of latitudes yield the same result.  Intuitively, I would expect only lat:33.38 should work.  Thanks for any help on this!

**Comment by lastzero** [comment]
We use S2 cell IDs internally: https://s2geometry.io/

Also a radius is assumed, as most users wouldn't expect exact matches down to the meter. The specific radius depends on where you are, it's also not 100% exact. Can you tell us more about your use case to get a better understanding?

**Comment by usul0184** [comment]
Sure, I'm looking for photos taken in a fairly small area, like within a city block.  It would be great if there was a way to specify the radius, but not sure if that is possible.  Thanks!

**Comment by lastzero** [comment]
Try the dist search filter: https://github.com/photoprism/photoprism/blob/develop/internal/query/photo_search.go#L378

**Comment by usul0184** [comment]
Thanks so much, this is very helpful!  A few suggestions if you agree:

1. Add documentation about the dist search.
2. Change GeoSearch.Dist from uint to float32 to allow more granular distances.
3. Eliminate SearchRadius completely; I think its presence is counter-intuitive without looking at the code.
4. This one is a little more involved and less important than the above, but maybe add the ability to search by true distance (e.g., in meters or kilometers).  The issue is that the dist search currently uses coordinates, and the true distance for longitude varies depending on what longitude you are on.

Again, thanks for this project; it is really amazing.
